exportlib.php

```diff
@@ -173,3 +174,167 @@ private function quote($str)
+                // Output the informations.
+                try {
+                    $courseobj = get_course($course->courseid);
+                    // Output courseinfos.
+                    $category = \core_course_category::get($courseobj->category);
+                    $catpath = explode('/', $category->path);
+                    $this->content .= $course->courseid . $seperator . $usercount . $seperator . $courseobj->shortname . $seperator;
+                    // TODO: Müssen wir das vorangehende '/' überprüfen? => davon hängt die Richtigkeit von $catpath[1] ab
+                    $this->content .= $courseobj->category . $seperator . $catpath[1] . $seperator;
+                } catch (Exception $ex) {
+                    // Output alternative courseinfos.
+                    $this->content .= $course->courseid . $seperator . $usercount . $seperator . ''
+                        . $seperator . '' . $seperator . '' . $seperator;
+                }
+
+                // Get amount of answers (for each answerpossibility) for the current question.
+                $params = array(
+                    "fid" => $this->feedbackid,
+                    "course" => $course->courseid,
+                    "qid" => $question->questionid
+                );
+                $sql = "SELECT
+			            answer,COUNT(*) AS count
+			        FROM
+			            {block_coursefeedback_answers}
+			        WHERE
+			            coursefeedbackid = :fid AND
+			            questionid = :qid AND
+			            course = :course
+			        GROUP BY
+			            answer";
+
+                // Initiate (reset) $anserres
+                $answerres = array();
+                if ($results = $DB->get_records_sql($sql, $params)) {
+                    foreach ($results as $answer) {
+                        // Save the amount of times the answeroption was chosen at the correspending index of the $answerres-array.
+                        $answerres[$answer->answer] = $answer->count;
+                    }
+                    block_coursefeedback_array_fill_spaces($answerres, 0, 8, 0);
+                } else {
+                    $answerres = array_fill(0, 8, 0);
+                }
+                // Vsum -> Amount of given Answers
+                $vsum = 0;
+                for($i = 1; $i <= 6; $i++) {
+                    $this->content .=  $answerres[$i] . $seperator;
+                    $vsum += $i * $answerres[$i];
+                }
+                $answercount = array_sum($answerres);
+                
+                // Ksum -> Amount of given Answers without the Amount of abstentions (abstentions were possible in earlier Versions).
+                $ksum    = $answercount - $answerres[0];
+                $average = $ksum > 0 ? ($vsum / $ksum) : 0;
+
+                // Output
+                $this->content .=  $average . $seperator . $answercount . $seperator . $answerres[0] . $newline;
+            }
```

Comment:
Das sollte am besten auch alles in einen Query - und dann vielleicht direkt mit der tablelib?

Reply:
Ergänzt in dbb0b3085b60846e7e459527b9638430a48d18e6
Siehe auch den Kommentar zum Entfernen der custom Buttons in der Rankingform oben.